app/src/main/java/de/westnordost/streetcomplete/quests/construction/MarkCompletedHighwayConstruction.java

```diff
@@ -0,0 +1,119 @@
+package de.westnordost.streetcomplete.quests.construction;
+
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.text.TextUtils;
+import android.util.Log;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import de.westnordost.osmapi.map.data.BoundingBox;
+import de.westnordost.streetcomplete.R;
+import de.westnordost.streetcomplete.data.meta.OsmTaggings;
+import de.westnordost.streetcomplete.data.osm.changes.StringMapChangesBuilder;
+import de.westnordost.streetcomplete.data.osm.download.MapDataWithGeometryHandler;
+import de.westnordost.streetcomplete.data.osm.download.OverpassMapDataDao;
+import de.westnordost.streetcomplete.data.osm.tql.OverpassQLUtil;
+import de.westnordost.streetcomplete.quests.DateHandler;
+import de.westnordost.streetcomplete.quests.YesNoQuestAnswerFragment;
+
+public class MarkCompletedHighwayConstruction extends MarkCompletedConstruction
+{
+	@Inject
+	public MarkCompletedHighwayConstruction(OverpassMapDataDao overpassServer) {
+		super(overpassServer);
+	}
+
+	@Override public boolean download(BoundingBox bbox, MapDataWithGeometryHandler handler)
+	{
+		return overpassServer.getAndHandleQuota(getOverpassQuery(bbox), handler);
+	}
+
+	/** @return overpass query string to get streets marked as under construction but excluding ones
+	 * - with invalid construction tag
+	 * - with tagged opening date that is in future
+	 * - recently edited (includes adding/updating check_date tags)
+	 * - with fixme (as in some situations this quest may add fixme note as an edit)
+	 * . */
+	private static String getOverpassQuery(BoundingBox bbox)
+	{
+		ArrayList<String> acceptedConstructionValues = new ArrayList<String>(Arrays.asList(OsmTaggings.ALL_ROADS));
+		acceptedConstructionValues.add("cycleway");
+		acceptedConstructionValues.add("footway");
+		acceptedConstructionValues.add("path");
+
+		String groupName = ".roads_under_construction";
+		String resultsName = ".roads_for_review";
+		return OverpassQLUtil.getGlobalOverpassBBox(bbox) +
+			getQueryPart("highway", groupName, 14) +
+			"(" +
+			"way[construction]" +
+			"[construction !~ \"^("+ TextUtils.join("|", acceptedConstructionValues)+")$\"]" +
+			") -> .invalid_construction_type;" +
+			"(" +
+			"  way[highway=construction][fixme];" +
+			") -> .with_fixme;" +
```

Comment:
Also, why are fixmes filtered out? (And why then only here, not in the other?)

Reply:
> So, if the access tag is not clear from context, the app could ask the surveyor about it after he answered "yes, the construction is finished!

It can be done as a separate issue but given very long list of access tags we run into tag translation issue. I can quite easily add it but it would add new translation string for every support access tag (currently list has 19 values).

Reply:
And it is also trickier than I initially though - what about situation where proper solution is not to accept all access tags or remove all of them, but to remove part of them?